components/datetime/src/provider.rs

```diff
@@ -180,22 +181,25 @@ impl DateTimeDates for structs::dates::gregory::DatesV1 {
         day_period: fields::DayPeriod,
         length: fields::FieldLength,
         hour: date::Hour,
+        minute: date::Minute,
     ) -> &Cow<str> {
-        let widths = match day_period {
-            fields::DayPeriod::AmPm => &self.symbols.day_periods.format,
-        };
+        use fields::{DayPeriod::NoonMidnight, FieldLength};
+        let widths = &self.symbols.day_periods.format;
         let symbols = match length {
-            fields::FieldLength::Wide => &widths.wide,
-            fields::FieldLength::Narrow => &widths.narrow,
+            FieldLength::Wide => &widths.wide,
+            FieldLength::Narrow => &widths.narrow,
             _ => &widths.abbreviated,
         };
-
-        //TODO: Once we have more dayperiod types, we'll need to handle
-        //      this logic in the right location.
-        if u8::from(hour) < 12 {
-            &symbols.am
-        } else {
-            &symbols.pm
+        match (day_period, u8::from(hour), u8::from(minute)) {
```

Comment:
Pure brainstorming: I'm wondering if we can do something to improve this matching. In my naive understanding, the term like `noon` or `midnight` is used in relation to the level of detail in the time presented.

If we're talking about hour+minute, then midnight is at 00:00, but if we are talking about just hour, then midnight is also 00:05.
Similarly if I'm formatting `12:00:05` then it is not `noon`, but if I'm formatting `12:00` then it is even if it is 5 seconds later.

Could we pass the information on the time tokens to be displayed to this function and use that to evaluate it?
Is it worth it?

Reply:
So what you're saying is we may want to achieve:

**(pseudo code)**
```rust
DateTime::try_from("2021-01-15T00:00:00")?.try_format("HH b")?       == "00 midnight"
DateTime::try_from("2021-01-15T00:00:00")?.try_format("HH:mm b")?    == "00:00 midnight"
DateTime::try_from("2021-01-15T00:00:00")?.try_format("HH:mm:ss b")? == "00:00:00 midnight"

DateTime::try_from("2021-01-15T00:00:01")?.try_format("HH b")?       == "00 midnight"
DateTime::try_from("2021-01-15T00:00:01")?.try_format("HH:mm b")?    == "00:00 midnight"
DateTime::try_from("2021-01-15T00:00:01")?.try_format("HH:mm:ss b")? == "00:00:01 AM"

DateTime::try_from("2021-01-15T00:01:00")?.try_format("HH b")?       == "00 midnight"
DateTime::try_from("2021-01-15T00:01:00")?.try_format("HH:mm b")?    == "00:01 AM"
DateTime::try_from("2021-01-15T00:01:00")?.try_format("HH:mm:ss b")? == "00:01:00 AM"

DateTime::try_from("2021-01-15T01:00:00")?.try_format("HH b")?       == "01 AM"
DateTime::try_from("2021-01-15T01:00:00")?.try_format("HH:mm b")?    == "01:00 AM"
DateTime::try_from("2021-01-15T01:00:00")?.try_format("HH:mm:ss b")? == "01:00:00 AM
```
Is that correct?

Reply:
I don't think day periods are very useful in h23/h24, but yeah. `12 midnight` and `12:00 midnight` but `12:00:01 am`.

Reply:
Did we decide to do this in this PR or follow up later?

Reply:
My understanding, based on a conversation in Slack, is that we still have not gained a clarity or consensus on what improvements we want to make here. 

@mihnita suggested that we should move this discussion to its own issue. 

@zbraniecki I think it's reasonable to land this now to avoid bitrot while we figure out what improvements ICU4X can make in this area.

I'm happy to create the issue for this.

Reply:
+1 to land without noon / midnight

Reply:
I think that is something that needs to be discussed then, even local to this GitHub issue, since that would mean not landing this PR at all.

Reply:
OK... reconsidering (discussed in meeting)

This is about patterns.
They are "supposed" to be "dumb", as in "do exactly what I'm telling you, even if bad i18n"
The real i18n issue will be in resolving the skeletons to patterns.

So OK to land as is (I will approve)